main/src/cgeo/geocaching/filters/core/GeocacheFilterType.java

```diff
@@ -32,7 +32,8 @@ public enum GeocacheFilterType {
     LOCATION("location", R.string.cache_filter_location, R.string.cache_filtergroup_details, LocationGeocacheFilter::new),
     STORED_LISTS("stored_list", R.string.cache_filter_stored_lists, R.string.cache_filtergroup_userspecific, StoredListGeocacheFilter::new),
     ORIGIN("origin", R.string.cache_filter_origin, R.string.cache_filtergroup_details, OriginGeocacheFilter::new),
-    STORED_SINCE("stored_since", R.string.cache_filter_stored_since, R.string.cache_filtergroup_userspecific, StoredSinceGeocacheFilter::new);
+    STORED_SINCE("stored_since", R.string.cache_filter_stored_since, R.string.cache_filtergroup_userspecific, StoredSinceGeocacheFilter::new),
+    LOGICAL_FILTER_GROUP(null, R.string.cache_filter_logical_filter_group, R.string.cache_filtergroup_special, AndGeocacheFilter::new);
```

Comment:
You reuse the LogicalGeocacheFilter in a way I had not foreseen or expected. This filter was meant to encapsulate logical OR/AND or NOT groups, not to encapsulate nested filters....

If this works it works. But given that here you instanciate always AND-Filters (naturally, LogicalGeocacheFilter is abstract) I wonder if this setup works when a nested filter is an OR filter?

(I would have expected that you create a new Filter class, e.g. "NestedGeocacheFilter", instead of extending/reusing LogicalGeocacheFilter)

Reply:
Actually, I was surprised as well. But I would like to stay with this solution, so we don't have to reinvent the wheel. I was just trying to build a GUI which can correctly display logical expressions which were technically already possible.

A different possible syntax would be something like this: `AND(filter1;nested:"AND(...)")`? But I guess, it's preferable to not blow up the expression unnecessarily...

>But given that here you instanciate always AND-Filters

True, this is necessary because `LogicalGeocacheFilter` itself is abstract and cannot be instantiated. Luckily, the Supplier is only used when adding a new filter, not when reading the existing config. This happens, cause the `ExpressionParser` handles these logical groups specially.

>I wonder if this setup works when a nested filter is an OR filter?

Yes, when the user selects OR inside the nested filter activity, an OR instance gets returned as filter and will be attached to the `LogicalFilterViewHolder` which can hold all types of logical groups. So yes, it works for OR filters as well...